src/components/Context/Context.tsx

```diff
@@ -0,0 +1,208 @@
+import { TypeOfFilter } from '../../types/TypeOfFilters';
+import { Error } from '../../types/TypeOfErrors';
+
+const USER_ID = 12049;
+
+interface AppContextTodoType {
+  USER_ID: number,
+  todos: Todo[],
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  filterType: TypeOfFilter,
+  setFilterType: React.Dispatch<React.SetStateAction<TypeOfFilter>>,
+  errors: Error | null,
+  setErrors: React.Dispatch<React.SetStateAction<Error | null>>,
+  tempTodo: Todo | null,
+  setTempTodo: React.Dispatch<React.SetStateAction<Todo | null>>,
+  isLoading: boolean,
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>,
+  filteredTodos: Todo[],
+  deleteTodoHandler: (value: number) => void,
+  addTodoTitle: (todo: Omit<Todo, 'id'>) => Promise<void>,
+  deletedIds: number[],
+  setDeletedIds: (value: number[]) => void,
+  updateTodos: (updatedTodo: Todo) => void,
+  toggleAllToCompletedTodos: () => void,
+}
+
+export const appContext = createContext<AppContextTodoType>({
+  USER_ID: 12049,
+  todos: [],
+  setTodos: () => { },
+  filterType: TypeOfFilter.All,
+  setFilterType: () => {},
+  errors: null,
+  setErrors: () => {},
+  tempTodo: null,
+  setTempTodo: () => {},
+  isLoading: false,
+  setIsLoading: () => {},
+  filteredTodos: [],
+  deleteTodoHandler: () => {},
+  addTodoTitle: async () => {},
+  deletedIds: [],
+  setDeletedIds: () => {},
+  updateTodos: () => {},
+  toggleAllToCompletedTodos: () => {},
+});
+
+type Props = {
+  children: ReactNode;
+};
+
+export const AppContextProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<TypeOfFilter>(TypeOfFilter.All);
+  const [errors, setErrors] = useState<Error | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [deletedIds, setDeletedIds] = useState<number[]>([]);
+
+  const filteredTodos = useMemo(() => {
+    switch (filterType) {
+      case TypeOfFilter.Active:
+        return todos.filter(todo => !todo.completed);
+      case TypeOfFilter.Completed:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  }, [todos, filterType]);
+
+  useEffect(() => {
+    TodoMethods.getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setErrors(Error.LoadTodos));
+  }, []);
+
+  const deleteTodoHandler = useCallback(
+    (todoId: number) => {
+      setDeletedIds((prevIds) => [...prevIds, todoId]);
+
+      setTimeout(() => {
+        setTodos(currentTodos => (
+          currentTodos.filter(todo => todo.id !== todoId)
+        ));
+
+        return TodoMethods.deleteTodo(todoId)
+          .then(() => setDeletedIds([]))
+          .catch(() => {
+            setErrors(Error.DeleteTodo);
+            setTodos(todos);
+            setDeletedIds(prevIds => (
+              prevIds.filter(prevId => prevId !== todoId)
+            ));
```

Comment:
You should filter loading ids in both cases, so move this action to finally block